tests/raftstore/test_transfer_leader.rs

```diff
@@ -117,3 +118,49 @@ fn test_node_pd_transfer_leader() {
     let mut cluster = new_server_cluster(0, 3);
     test_pd_transfer_leader(&mut cluster);
 }
+
+fn test_transfer_leader_during_snapshot<T: Simulator>(cluster: &mut Cluster<T>) {
+    let pd_client = cluster.pd_client.clone();
+    // Disable default max peer count check.
+    pd_client.disable_default_rule();
+    cluster.cfg.store_cfg.raft_log_gc_tick_interval = 20;
+    cluster.cfg.store_cfg.raft_log_gc_limit = 2;
+
+    let r1 = cluster.run_conf_change();
+    pd_client.must_add_peer(r1, new_peer(2, 2));
+
+    for i in 0..1024 {
+        let key = format!("{:01024}", i);
+        let value = format!("{:01024}", i);
+        cluster.must_put(key.as_bytes(), value.as_bytes());
+    }
+
+    // hook transport and drop all snapshot packet, so follower's status
+    // will stay at snapshot.
+    cluster.hook_transport(DropSnapshot);
+    pd_client.must_add_peer(r1, new_peer(3, 3));
+    // a just added peer needs wait a couple of ticks, it'll communicate with leader
+    // before getting snapshot
+    sleep_ms(1000);
+
+    cluster.transfer_leader(r1, new_peer(2, 2));
+    cluster.reset_transport_hooks();
```

Comment:
leader transfer doesn't happen before reset, not because we add the DropSnapshot packet filter but because there are followers in snapshot status.
those have nothing to do with DropSnapshot itself, so why should we check before and after reset? 
or, do you mean transfer leader again and check it can success after reset?